conjure-java-jackson-serialization/src/test/java/com/palantir/conjure/java/serialization/ObjectMappersTest.java

```diff
@@ -128,11 +131,115 @@ public void testLongTypeDeserializationFromString() throws IOException {
         assertThat(MAPPER.readValue("\"1\"", Long.TYPE)).isEqualTo(1L);
     }
 
+    @Test
+    public void testLongBeanTypeDeserializationFromString() throws IOException {
+        assertThat(MAPPER.readValue("{\"value\":\"1\"}", LongBean.class)).isEqualTo(new LongBean(1L));
+    }
+
+    @Test
+    public void testLongBeanTypeDeserializationFromNumber() throws IOException {
+        assertThat(MAPPER.readValue("{\"value\":\"1\"}", LongBean.class)).isEqualTo(new LongBean(1L));
+    }
+
+    static final class LongBean {
```

Comment:
can we use something we generate from Conjure here? or an immutable here

Reply:
Conjure is not impacted by the bug, this requires direct bean field access via afterburner. Immutables generated objects may work, but there's no guarantee around their implementation and it may change between versions.

Reply:
ok, I think that having class definitions (especially verbose ones) scattered throughout the test class makes it harder to read. Could we either push these to the bottom or pull them out into separate files

Reply:
I suppose, I tend to prefer having classes for tests colocated with the test so it's easy to quickly grok the entire case, but we can put them at the bottom of the class ¯\_(ツ)_/¯

Reply:
ok I don't have strong feelings here